src/jit/lsra.cpp

```diff
@@ -6868,7 +6873,7 @@ LinearScan::resolveLocalRef(GenTreePtr treeNode, RefPosition * currentRefPositio
     if (currentRefPosition->registerAssignment == RBM_NONE)
     {
         // Either requires no register or reg is optional.
-        assert(!currentRefPosition->IsActualRef() ||
+        assert(!currentRefPosition->RequiresRegister() ||
                currentRefPosition->IsRegOptional());
```

Comment:
I believe that the second clause is now redundant (which is as it should be, since it really makes no sense for `RequiresRegister()` to be true, but `IsRegOptional()` to also be true).